apps/workspace-host/src/interface.ts

```diff
@@ -859,64 +905,67 @@ async function initSequence(workspace_id, useInitialZip, res) {
     await sqldb.queryOneRowAsync(sql.select_workspace, { workspace_id })
   ).rows[0];
 
-  const workspace = {
-    id: workspace_id,
-    course_id,
-    institution_id,
-    version,
-    launch_uuid: uuid,
-    local_name: `workspace-${uuid}`,
-    remote_name: `workspace-${workspace_id}-${version}`,
-  };
-
   logger.info(`Launching workspace-${workspace_id}-${version} (useInitialZip=${useInitialZip})`);
   try {
     // Only errors at this level will set host to unhealthy.
 
-    try {
-      debug('init: configuring workspace');
-      workspace.launch_port = await _allocateContainerPort(workspace);
-      workspace.settings = await _getWorkspaceSettings(workspace.id);
-    } catch (err) {
-      logger.error(`Error configuring workspace ${workspace_id}`, err);
-      safeUpdateWorkspaceState(
-        workspace_id,
-        'stopped',
-        'Error configuring workspace. Click "Reboot" to try again.',
-      );
-      return; // don't set host to unhealthy
-    }
+    const workspace = await run(async () => {
+      try {
+        debug('init: configuring workspace');
+        return {
+          id: workspace_id,
+          course_id,
+          institution_id,
+          version,
+          launch_uuid: uuid,
```

Comment:
What is the convention here? I thought for JS we should use camelCase not snake_case. But the ids coming in from the database are snake_case.

Reply:
🙈 generally, things that come from or go to the database (and often data coming from inputs on the client) use `snake_case`. We are not religious about this. I believe the workspace host was written by an individual who had a penchant for `snake_case`, and so there's perhaps an excess amount of that here.

Matt and I both agree that Ruby/Python/Rust nailed it with their mix of `PascalCase` and `snake_case` lol.